### PR TITLE
Code improvements

### PR DESCRIPTION
**Pull request by rster2002**
# Goal
This PR focuses on structural changes and defining a standard moving forward. Making these changes should allow people to implement code changes more easily without having to make changes in multiple (unrelated files).

## Implementation idea

MicrosoftTeams-image (1)

The bottom model was chosen. The idea is to split the game into three parts:

- [x] A global state for score and time etc.
- [x] Instances: these would be the actual game objects
- [x] Controllers: these orchestrate the instances and work together to form the logic for the game.

### CODE REVIEW

**Comment by rster2002** [comment]
We might want to opt for renaming `models` to `instances` as I would like to reserve Models data models/classes.

File: Demo/Demo/Assets/Scripts/LevelController.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Linq;
+using UnityEngine;
+
+namespace DefaultNamespace
+{
+    public class LevelController : MonoBehaviour
```
Comment (rster2002) [inline Demo/Demo/Assets/Scripts/LevelController.cs:7]: Is this something we want to make abstract? I don't know how many different levels we would have besides `tutorial` and `normal` level?

Comment (suchmememanyskill) [reply Demo/Demo/Assets/Scripts/LevelController.cs:7]: As discussed today, We could make it a scriptable object. I think it's outside of the scope for now though

**Review by rster2002 — COMMENTED** [review]
Cannot actually approve because I created the PR, but it's good to go 👍

**Review by suchmememanyskill — APPROVED** [review]

**Pull request merged**